src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import { DispatchContext } from '../../Store';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { id, title, completed } = todo;
+
+  const dispatch = useContext(DispatchContext);
+  const [isReNaming, setIsReNaming] = useState(false);
```

Comment:
```suggestion
  const [isRenaming, setIsRenaming] = useState(false);
```

Reply:
what do you mean about handle plural?
looks right!